### ISSUE TITLE
Validate `Job::dispatch()` arguments against constructor signature

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#698]
Arguments passed to `SomeJob::dispatch()` should be validated against the job's `__construct()` signature.

```php
class SendEmail {
    public function __construct(private User $user, private string $subject) {}
}

SendEmail::dispatch($user);           // missing $subject — should warn
SendEmail::dispatch('oops', 'Hello'); // wrong type — should warn
```

Larastan implements this via `DispatchableRule`.

**Implementation:** `AfterExpressionAnalysisInterface` on `Dispatchable::dispatch()` static calls — resolve the declaring class constructor and validate argument count and types.
